utils/hare-start-node

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+set -eu -o pipefail
+# set -x
+export PS4='+ [${BASH_SOURCE[0]##*/}:${LINENO}${FUNCNAME[0]:+:${FUNCNAME[0]}}] '
+
+PROG=${0##*/}
+
+. hare-utils # import utility functions
+
+die() {
+    echo "$PROG: $*" >&2
+    exit 1
+}
+
+say() {
+    echo -n "$(date '+%F %T'): $*"
+}
+
+is_server_node() {
+    get_server_nodes | grep -w $(node-name) > /dev/null
+}
+
+if sudo systemctl --quiet is-active hare-consul-agent; then
+    die 'hare-consul-agent is active ==> cluster is already running'
+fi
+
+say 'Starting Consul agent on this node...'
+sudo systemctl start hare-consul-agent
+
+# Wait for Consul's internal leader to be ready.
+# (Until then the KV store won't be accessible.)
+# Here we are checking if any of the peer nodes is already elected as the
+# leader. 
+while [[ ! $(get_leader) ]]; do
+    sleep 1
+    echo -n '.'
+done
+echo ' OK'
+
+if is_server_node; then
+    # Starting phase 1 (confd servers)
+    start_motr m0d phase1
+fi
+
+# Starting phase 2 (ios)
+start_motr m0d phase2
+
+. update-consul-conf --dry-run  # import S3_IDs
+if [[ -n $S3_IDs ]]; then
```

Comment:
Don't think we need to import S3_IDS here and check them, just invoke start_s3server, bootstrap-node already fetches IDs and start the processes accordingly.

Reply:
**Done.**
Invoked start_s3server() directly, but since in `bootstrap-node` if the service id is not present, it simply exits. Hence, added a check of `Service not found`.  

![image](https://user-images.githubusercontent.com/67725934/108992218-10f11c00-76bf-11eb-83e1-4eea43bab99a.png)

Reply:
I think it is okay to exit silently if s3servers are not part of the configuration. Throwing a `Service not found` error may confuse the user as he may not be able to distinguish if its an error or something that is expected.

Reply:
> start_s3server() {
    say 'Starting S3 servers (phase3)...'
    bootstrap-node --phase phase3 &
    pids=($!)
    wait4 ${pids[@]}
    echo ' OK'
}

According, to function we do log this msg `'Starting S3 servers (phase3)...'` , now if S3 IDs are missing, it wont start the services but still it will log the message of starting services. This will mislead the user. Is that okay?

Reply:
@Shreya-18 I think instead of service not found, "No S3 servers mentioned for this config" or somewhat similar message would be fine. I mean, Starting...then searching... and then no s3 servers available for this configuration. Just a suggestion.

Reply:
For me the message "Starting S3 servers (phase3)... Service not found OK" is super misleading 😁 
Which service not found? How can I fix that? And if it is a problem, why OK is written? If everything is OK, then why do I see this message?

I would suggest just not print the message about S3 servers if they are not expected to be started. Otherwise the message is just a noise.

Reply:
@Shreya-18 , I think I had asked you earlier to invoke `start_s3server()` unconditionally, somehow I thought bootstrap-node skips s3servers if not present and also will not print the message but looks like the message is printed earlier. So let's revert the S3_IDS conditional check before invoking `start_s3server()`.

Reply:
As discussed,  bootstrap-node does skip s3servers but the message is still printed.  Having a check before invoking start_S3servers() is better. 

Hence, reverting back.

Reply:
Done